src/webnn/native/dml/ContextDML.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 The WebNN-native Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef WEBNN_NATIVE_DML_CONTEXT_DML_H_
+#define WEBNN_NATIVE_DML_CONTEXT_DML_H_
+
+#include "webnn/native/Context.h"
+
+#include "common/Log.h"
+#include "dml_platform.h"
+#include "webnn/native/Graph.h"
+
+namespace webnn::native::dml {
+
+    class Context : public ContextBase {
+      public:
+        static ContextBase* Create(ComPtr<IDXGIAdapter1> adapter, bool useDebugLayer);
+        ~Context() override = default;
+
+      private:
+        Context(ComPtr<IDXGIAdapter1> adapter, bool useDebugLayer);
+        HRESULT Initialize();
+
+        GraphBase* CreateGraphImpl() override;
+
+        ComPtr<IDMLDevice> mDevice;
+        ComPtr<ID3D12Device> mD3D12Device;
+        ComPtr<IDMLCommandRecorder> mCommandRecorder;
```

Comment:
What do you think about having `CommandRecorderDML`? 

`CommandRecorderDML` would store the`ID3D12CommandAllocator` and be stored within _this_ context. Then everything that needs the recorder but not the execution context, eg. InitializeOperator, ExecuteOperator is moved to `CommandRecorderDML`. This keeps the logic of recording clearly and cleanly separated from execution. The context just asks for the recorded command-list from this instance to execute.

Reply:
Do you mean move these member variables into 
```
CommandRecorderDML{
        ComPtr<IDMLDevice> mDevice;
        ComPtr<ID3D12Device> mD3D12Device;
        ComPtr<IDMLCommandRecorder> mCommandRecorder;
        ComPtr<ID3D12CommandQueue> mCommandQueue;
        ComPtr<ID3D12CommandAllocator> mCommandAllocator;
        ComPtr<ID3D12GraphicsCommandList> mCommandList;
}
```
?

Reply:
Yup, that's right.